mne/preprocessing/eyetracking/eyetracking.py

```diff
@@ -5,7 +5,100 @@
+    raw : instance of Raw
+        The raw data with at least one ``'pupil'`` or ``'eyegaze'`` channel.
+    buffer : float | array-like of float, shape ``(2,))``
+        The time in seconds before and after a blink to consider invalid and
+        include in the segment to be interpolated over. Default is ``0.05`` seconds
+        (50 ms). if array-like, the first element is the time before the blink and the
+        second element is the time after the blink to consider invalid, for example,
+        ``(0.025, .1)``.
+    match : str | list of str
+        The description of annotations to interpolate over. If a list, data within
+        annotations matching any of the strings in the list will interpolated over.
+        Defaults to ``'BAD_blink'``.
+
+    Returns
+    -------
+    self : instance of Raw
+        Returns the modified instance.
+    """
+    _check_preload(raw, "interpolate_blinks")
+    _validate_type(raw, BaseRaw, "raw")
+    _validate_type(buffer, (float, tuple, list, np.ndarray), "buffer")
+    _validate_type(match, (str, tuple, list, np.ndarray), "match")
+
+    # determine the buffer around blinks to include in the interpolation
+    buffer = np.array(buffer, dtype=float)
+    if buffer.size == 1:
+        buffer = np.array([buffer, buffer])
+
+    if isinstance(match, str):
+        match = [match]
+
+    # get the blink annotations
+    blink_annots = [annot for annot in raw.annotations if annot["description"] in match]
+    if not blink_annots:
+        warn("No annotations matching {} found. Aborting.".format(match))
+        return raw
+    _interpolate_blinks(raw, buffer, blink_annots)
+
+    # remove bad from the annotation description
+    for desc in match:
+        if desc.startswith("BAD_"):
+            logger.info("Removing 'BAD_' from {}.".format(desc))
+            raw.annotations.rename({desc: desc.replace("BAD_", "")})
+    return raw
+
+
+def _interpolate_blinks(raw, buffer, blink_annots):
+    """Interpolate eyetracking signals during blinks in-place."""
+    from scipy.interpolate import interp1d
+
+    logger.info("Interpolating missing data during blinks...")
+    pre_buffer, post_buffer = buffer
+    # iterate over each eyetrack channel and interpolate the blinks
+    for i, ch_info in enumerate(raw.info["chs"]):
+        if ch_info["kind"] != FIFF.FIFFV_EYETRACK_CH:
+            continue
+        # Create an empty boolean mask
+        mask = np.zeros_like(raw.times, dtype=bool)
+        for annot in blink_annots:
+            if "ch_names" not in annot or not annot["ch_names"]:
+                msg = "blink annotation missing 'ch_names' key. got: {}".format(annot)
+                raise ValueError(msg)
+            start = annot["onset"] - pre_buffer
+            end = annot["onset"] + annot["duration"] + post_buffer
+            ch_names = annot["ch_names"]
+            if ch_info["ch_name"] not in ch_names:
+                continue  # skip if the channel is not in the blink annotation
```

Comment:
I don't really work with annotations much, so please excuse my ignorance here ... I was wondering:
a) is there any scenario where annotations would only exist for one channel? E.g. based on how additional preproc was done?
b) if a), is there an easy way to "broadcast" annotations to other channels?
c) if not b), should we allow some way to include channels and treat them based on the annotation of other channels?

Reply:
> is there any scenario where annotations would only exist for one channel? 

The EyeLink data files can specify "blink_L" / "blink_R" / "blink_BOTH" when recording in binocular mode. Accordingly, our reader has been updated to annotate L and R channels separately or jointly.

> is there an easy way to "broadcast" annotations to other channels?

No. `annotations.ch_names` must either be `None` (all annotations are channel-agnostic) or a (ragged) array of channel names (all annotations are channel-specific; channel-agnostic spans will be a list of all channel names).

> should we allow some way to include channels and treat them based on the annotation of other channels?

My hunch here is "YAGNI", WDYT?

Reply:
Thanks for the explanation @drammock, now I understand it better I think!
Mh ... what if you have "blink_L" but you want to interpolate the data in left _and_ right channels?

Reply:
> Mh ... what if you have "blink_L" but you want to interpolate the data in left and right channels?

I would say "let's wait until someone wants to", or I would tell the person in this case to modify the list of channels annotated by the BAD_blink annotation. I think in practice if the person really did blink one eye, the other eye is open and doesn't need to be interpolated. So even if single-eye blinks are uncommon, we cover the "common" way this (uncommon) data should be processed I think

Reply:
Probably -- worth trying. Also worth double checking that my understanding of the `ch_names` is correct...

But for this PR I think it doesn't matter too much. @britta-wstnr okay with adding / extending / figuring out how to deal with "I want to interpolate more channels than what the channel-aware annotations currently say are bad" later?

Reply:
> But for this PR I think it doesn't matter too much. @britta-wstnr okay with adding / extending / figuring out how to deal with "I want to interpolate more channels than what the channel-aware annotations currently say are bad" later?

Yes, yes, absolutely! This was just a question that formed in my mind while going over the code 🙂